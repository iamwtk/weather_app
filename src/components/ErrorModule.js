import React from 'react'



import style from './errorModule.scss'

const ErrorModule = (props) => (
  <div className={style.error_module}>
  <p>Sorry, there was an error!</p>
  {props.errors.map((err, i) => <span key={i}>{err.message}</span>)}
  </div>
);

export default ErrorModule;
