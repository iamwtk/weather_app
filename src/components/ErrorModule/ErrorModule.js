import React, { PropTypes } from 'react'

import style from './errorModule.scss'

/**
 * <ErrorModule />
 * displays errors from array passed as props
 * @param {array} errors array of errors occured while fetching
 */
const ErrorModule = ({errors}) => (
  <div className={style.error_module}>
    <p>Sorry, there was an error!</p>
    {errors.map((err, i) => <span key={i}>{err.message}</span>)}
  </div>
)
export default ErrorModule
