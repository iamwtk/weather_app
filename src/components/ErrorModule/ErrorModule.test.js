import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';

import ErrorModule from './ErrorModule'
import style from './errorModule.scss'

describe('<ErrorModule/>', () => {
  let component
  const errors = [
    {message: "error 1"},
    {message: "error 2"}
  ]
  
  beforeEach(() => {
    component = shallow(<ErrorModule errors={errors}/>)
  })

  describe('Snapshot', () => {
    it('should match the snapshot', () => {
      const tree = toJson(component)
      expect(tree).toMatchSnapshot()
    })
  })
})
