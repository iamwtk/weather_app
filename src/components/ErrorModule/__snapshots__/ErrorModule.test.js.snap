// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorModule/> Snapshot should match the snapshot 1`] = `
<div
  className="error_module"
>
  <p>
    Sorry, there was an error!
  </p>
  <span
    key="0"
  >
    error 1
  </span>
  <span
    key="1"
  >
    error 2
  </span>
</div>
`;
